type-generator/src/main/java/io/mesosphere/types/dtr/compiler/TypeCompiler.java

```diff
@@ -0,0 +1,30 @@
+package io.mesosphere.types.dtr.compiler;
+
+import io.mesosphere.types.dtr.compiler.internal.CompilerEngine;
+import io.mesosphere.types.dtr.repository.internal.RepositoryView;
+
+/**
+ * Type compiler generates the sources for the given types
+ */
+public class TypeCompiler {
+
+    /**
+     * The engine to use for compiling the type
+     */
+    private CompilerEngine engine;
+
+    /**
+     * The repository view from which to load the compiler templates
+     */
+    private RepositoryView compilerView;
+
+    /**
+     * Initialize the type compiler view
+     * @param engine
+     * @param compilerView
+     */
+    public TypeCompiler(CompilerEngine engine, RepositoryView compilerView) {
```

Comment:
Why is this compiler not iterating over the types?

Reply:
I wanted to add a bit more flexibility on the compiler interface, allowing the user to perform fine-grained, per-item compilation.

But I would guess that adding the support to additionally compiling the entire scope would make sense :+1: